tests/Integration/Behaviour/Features/Context/Domain/OrderFeatureContext.php

```diff
@@ -1356,4 +1367,25 @@ private function getOrderDiscountByName(int $orderId, string $cartRuleName): ?Or
 
         return null;
     }
+
+    /**
+     * @param int $orderId
+     *
+     * @return TaxCalculator|null
+     *
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    private function getOrderTaxCalculator(int $orderId)
+    {
+        $order = new Order($orderId);
+        $taxes = $order->getOrderDetailTaxes();
```

Comment:
I didn't know this one! Seems handy!
But it means each product can have its dedicated tax group?

Reply:
Given the way it works it seems to behave as if it was a possibility, but I'm not sure in what context can this happen 🤔